### PR TITLE
feat(popover): add new component popover

### PR DESCRIPTION
**Pull request by skhamvon**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build#124.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#128.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#133.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#140.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#142.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#143.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#159.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#164.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#178.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#185.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#187.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#189.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#191.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#192.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#196.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#220.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build-storybook#221.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build-storybook#221.1 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build-storybook#222.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build-storybook#224.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build-storybook#224.1 ✘  
* 
  * build ✘

File: packages/design/assets/design-images/popover/popover-preview.png
Comment (astagnol) [inline packages/design/assets/design-images/popover/popover-preview.png:1]: Maybe we should update this png with current colors for typography

File: packages/stencil/components/popover/src/components/osds-popover-content/osds-popover-content.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+@import 'styles/osds-popover-content.color';
+@import 'styles/osds-popover-content.size';
+@import 'styles/osds-popover-content.typography';
+@import '~@ovhcloud/ods-theming/color/ods-theming-color';
+@import '~@ovhcloud/ods-theming/ods-theme';
+
+// CSS targeted only for this component
+// /!\ for theming purposes, it has to be done in theming files
+// (i.e. packages/libraries/theming/...)
+:host {
```
Comment (astagnol) [inline packages/stencil/components/popover/src/components/osds-popover-content/osds-popover-content.scss:11]: Maybe we can cleaned unused css

File: packages/stencil/components/popover/src/components/osds-popover-content/osds-popover-content.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+@Component({
+  tag: 'osds-popover-content',
+  styleUrl: 'osds-popover-content.scss',
+  shadow: true
+})
+export class OsdsPopoverContent implements OdsPopoverContent<OdsStencilMethods<OdsPopoverContentMethods>, OdsStencilEvents<OdsPopoverContentEvents>> {
+  controller: OdsPopoverContentController = new OdsPopoverContentController(this);
+  @Element() el!: HTMLElement;
+
+  // Component properties as @Prop
```
Comment (astagnol) [inline packages/stencil/components/popover/src/components/osds-popover-content/osds-popover-content.tsx:22]: to remove (from generator but no need to keep it there)

File: packages/stencil/components/popover/src/components/osds-popover/styles/osds-popover.color.scss [hunk 3]
```diff
@@ -0,0 +1,33 @@
+@import '~@ovhcloud/ods-theming/color/ods-component-color';
+@import './osds-popover.mixins';
+
+@mixin osds-popover-theme-color-variant-default() {
+  @include ods-and-all-hue-foreach-theme-color((
+          color: '800',
+          background-color: '100'
+  )) using($colors) {
+    @include osds-popover-on-selected-host() {
+      //color: map_get($colors, color);
```
Comment (astagnol) [inline packages/stencil/components/popover/src/components/osds-popover/styles/osds-popover.color.scss:10]: could be removed

File: packages/stencil/components/popover/src/components/osds-popover/styles/osds-popover.typography.scss [hunk 4]
```diff
@@ -0,0 +1,16 @@
+@import '~@ovhcloud/ods-theming/typography/ods-theming-typography';
+@import '~@ovhcloud/ods-theming/typography/ods-component-typography';
+@import './osds-popover.mixins';
+
+// Main CSS mixin for typography
+@mixin osds-popover-theme-typography() {
+  :host([size='md']) .popover__wrapper {
+    //$typography-properties: ods-get-typography-properties(body, '400');
```
Comment (astagnol) [inline packages/stencil/components/popover/src/components/osds-popover/styles/osds-popover.typography.scss:8]: if not necessary, can be removed

**Review by astagnol — CHANGES_REQUESTED** [review]

File: packages/libraries/core/src/components/popover/popover/ods-popover-controller.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  handleTriggerClick(): void {
+    this.logger.log('Click on trigger');
+
+    if (!this.component.surface) {
+      return;
+    }
+    this.component.surface.opened = !this.component.surface.opened;
+  }
+
+  handleTriggerKey(event: KeyboardEvent): void {
```
[the remark this answers is not in the document]
Comment (skhamvon) [reply packages/libraries/core/src/components/popover/popover/ods-popover-controller.ts:32]: As seen together, there is only button as trigger now, and it work with space and Enter

File: packages/design/assets/design-images/popover/popover-preview.png
Comment (skhamvon) [reply packages/design/assets/design-images/popover/popover-preview.png:1]: Updated

**Comment by ovh-cds** [comment]
CDS Report build#391.0 ■   
* 
  * build ■

**Comment by ovh-cds** [comment]
CDS Report build#391.0 ■   
* 
  * build ■

**Comment by ovh-cds** [comment]
CDS Report build-storybook#392.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#394.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#393.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#399.0 ■   
* 
  * build ■

**Comment by ovh-cds** [comment]
CDS Report build#399.0 ■   
* 
  * build ■

**Comment by ovh-cds** [comment]
CDS Report build#405.0 ■   
* 
  * build ■

**Comment by ovh-cds** [comment]
CDS Report build#405.0 ■   
* 
  * build ■

**Comment by ovh-cds** [comment]
CDS Report build-storybook#424.0 ✘  
* 
  * build ✘

File: packages/libraries/core/src/components/popover/popover-content/ods-popover-content-controller.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  // private readonly logger = new OdsLogger('OdsPopoverContentController');
+
+  constructor(component: OdsPopoverContent) {
+    super(component);
+  }
+
+  /**
+   * Attributes validation documentation
+   */
+  validateAttributes(): void {
```
[the remark this answers is not in the document]
Comment (skhamvon) [reply packages/libraries/core/src/components/popover/popover-content/ods-popover-content-controller.ts]: Removed

**Comment by ovh-cds** [comment]
CDS Report build#764.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#775.0 ✘  
* 
  * build ✘

File: packages/libraries/core/jest.config.ts [hunk 7]
```diff
@@ -19,4 +19,5 @@ const args = process.argv.slice(2);
  * ```
  */
 const config: Config.InitialOptions = OdsGetJestConfig({ basePath: '<rootDir>/../../..', stencil: false, args });
+console.log(config);
```
Comment (dpellier) [inline packages/libraries/core/jest.config.ts:22]: should be removed

File: packages/libraries/core/src/components/popover/popover/ods-popover-controller.ts [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  /**
+   * Handle click event on Surface
+   */
+  handleSurfaceKey(event: KeyboardEvent): void {
+    if (event.key === "Escape") {
+      this.logger.log('EscapeKey in surface');
+      if(this.component.surface) {
+        this.component.surface.opened = false;
```
Comment (dpellier) [inline packages/libraries/core/src/components/popover/popover/ods-popover-controller.ts:63]: should use inner method `closeSurface`

File: packages/libraries/core/src/components/popover/popover/ods-popover-controller.ts [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+      }
+    }
+  }
+
+  /**
+   * Handle click event on document and check if it is not on the component itself
+   */
+  checkForClickOutside(event: any): void {
```
Comment (dpellier) [inline packages/libraries/core/src/components/popover/popover/ods-popover-controller.ts:71]: can be removed ? should be managed by the surface

File: packages/libraries/core/src/components/popover/popover/ods-popover-controller.ts [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    } else {
+      this.logger.log('Click outside component while it is opened');
+      this.component.surface.opened = false;
+    }
+  }
+
+  /**
+   * Method to close the surface
+   */
+  closeSurface(): void {
```
Comment (dpellier) [inline packages/libraries/core/src/components/popover/popover/ods-popover-controller.ts:83]: there is already one in the surface object

**Review by dpellier — CHANGES_REQUESTED** [review]
Regarding keyboard navigation, I couldn't focus the trigger on the storybook, so impossible to test Enter/Escape behavior

**Review by dpellier — APPROVED** [review]

**Review by astagnol — APPROVED** [review]

**Pull request merged**